web/apps/serve.go

```diff
@@ -313,6 +307,12 @@ func ServeAppFile(c echo.Context, i *instance.Instance, fs appfs.FileServer, web
 	sessID := ""
 	if isLoggedIn {
 		sessID = sess.ID()
+	} else if file == "" || file == route.Index {
+		if !route.Public {
+			if handled, err := middlewares.CheckOAuthClientsLimitExceeded(c); handled {
```

Comment:
I don't understand. The PR description says:

> Therefore, we'll make sure the redirection to the paywall only occurs when the user is logged in

But, here, we check for the OAuth clients limit only if the user is NOT logged in. What am I missing?

Reply:
You're not missing anything. That's my mistake :man_facepalming:.
I don't know how my manual test worked. I'll fix this and run some more tests.